Remotion/Validation/UnitTests/Rules/ObjectValidationRuleTest.cs

```diff
@@ -1,21 +1,176 @@
+
+      var rule = new ObjectValidationRule<object>(null, new[] { validator1Stub.Object, validator2Stub.Object });
+
+      var result = rule.Validate(context);
+      Assert.That(result, Is.EquivalentTo(new[] { failure1, failure2 }));
+    }
+
+    [Test]
+    public void Validate_UsesCorrectChildContextToBuildParentContext ()
+    {
+      var instance = new object();
+      var validationContext = new ValidationContext(instance);
+
+      var failure1 = new ObjectValidationFailure(
+          15,
+          "15 is not a number",
+          "Localized validation message: 15 is not a number.");
+
+      var validator1Stub = new Mock<IObjectValidator>();
+      validator1Stub
+          .Setup(_ => _.Validate(It.Is<ObjectValidatorContext>(ctx => ctx.ParentContext == validationContext && ctx.Instance == instance)))
+          .Returns(new[] { failure1 });
+
+      var rule = new ObjectValidationRule<object>(null, new[] { validator1Stub.Object });
+
+      var result = rule.Validate(validationContext);
+      Assert.That(result, Is.EquivalentTo(new[] { failure1 }));
+    }
+
+    [Test]
+    public void Validate_WithNoValidators_ReturnsEmptyResult ()
+    {
+      var context = new ValidationContext(new object());
+      var rule = new ObjectValidationRule<object>(null, new List<IObjectValidator>());
+
+      var result = rule.Validate(context);
+
+      Assert.That(result, Is.Empty);
+    }
+
+    [Test]
+    public void Validate_WithNullAsInstanceToValidate_ReturnsEmptyResultAsValidatorsAreNotCalled ()
+    {
+      var context = new ValidationContext(null);
+
+      var validatorStub = new Mock<IObjectValidator>();
+      validatorStub
+          .Setup(_=>_.Validate(It.IsAny<ObjectValidatorContext>()))
+          .Returns(new [] {new ObjectValidationFailure(
+                              15,
+                              "15 is not a number",
+                              "Localized validation message: 15 is not a number.")});
+
+      var rule = new ObjectValidationRule<object>(null, new [] {validatorStub.Object});
+
+      var result = rule.Validate(context);
+
+      Assert.That(result, Is.Empty);
+    }
+
+    [Test]
+    public void Validate_WithUnfulfilledCondition_ReturnsEmptyResultAsValidatorsAreNotCalled ()
+    {
+      Func<object, bool> condition = _ => false;
+      var context = new ValidationContext(new object());
+
+      var validatorStub = new Mock<IObjectValidator>();
+      validatorStub
+          .Setup(_=>_.Validate(It.IsAny<ObjectValidatorContext>()))
+          .Returns(new [] {new ObjectValidationFailure(
+              15,
+              "15 is not a number",
+              "Localized validation message: 15 is not a number.")});
+
+      var rule = new ObjectValidationRule<object>(condition, new [] {validatorStub.Object});
+
+      var result = rule.Validate(context);
+
+      Assert.That(result, Is.Empty);
+    }
+
+    [Test]
+    public void IsActive_WithoutInstanceToValidate_ReturnsFalse ()
```

Comment:
I think we should make the condition null in this test. WithoutInstanceToValidate has nothing to do with the consditionand making the condition null is more explcit aobut this. This should be updatd, please, but I don't need to see it after.

Reply:
Problem here is that this test is about `IsActive,` which always returns true given a null condition. You may have thought that this is the `Validate` method?

Given this, we need the condition to return true to actually show a different output given the missing instance to validate.

Reply:
oh, yeah. thaks. gotcha! no change then